lib/index.js

```diff
@@ -0,0 +1,126 @@
+const fs = require('fs');
+const {
+  fetchLabels,
+  createLabels,
+  deleteLabels,
+  updateLabels,
+} = require('./api');
+
+async function labelControl() {
+  const labelData = await fetchLabels();
+  const formattedPublishedLabels = await labelFormatter(labelData);
+  const data = fs.readFileSync('./label-config.json', { encoding: 'utf8' });
+
+  const localConfig = JSON.parse(data.toString());
+  const illegalLabels = checkForSameLabelName(
+    localConfig,
+    formattedPublishedLabels
+  );
+  if (illegalLabels) {
+    throw new Error(
+      'The config has labels with the same name. The Label names have to be unique.'
+    );
+  }
+  const newLabels = checkForNewOrOldLabels(
+    localConfig,
+    formattedPublishedLabels
+  );
+  const labelsToDelete = checkForNewOrOldLabels(
+    formattedPublishedLabels,
+    localConfig
+  );
+  const labelsToUpdate = checkForModifications(
+    localConfig,
+    formattedPublishedLabels
+  );
+  if (newLabels.length !== 0) {
+    await createLabels(newLabels);
+  }
+  if (labelsToDelete.length !== 0) {
+    await deleteLabels(labelsToDelete);
+  }
+  if (labelsToUpdate.length !== 0) {
+    await updateLabels(labelsToUpdate);
+  }
+  await writeLabelsConfig();
+  const changes =
+    newLabels.length + labelsToDelete.length + labelsToUpdate.length;
+  const summary =
+    changes !== 0
+      ? {
+          message: 'Labels synchronized!',
+          stats: {
+            created: newLabels.length,
+            deleted: labelsToDelete.length,
+            updated: labelsToUpdate.length,
+          },
+        }
+      : { message: 'Labels already up to date!' };
+  return summary;
+}
+
+async function writeLabelsConfig() {
+  const labelData = await fetchLabels();
+  const labels = await labelFormatter(labelData);
+  fs.writeFileSync('./label-config.json', JSON.stringify(labels, null, 2));
+}
+
+function checkForNewOrOldLabels(newLabels, oldLabels) {
+  const updatedLabels = [];
+  const OldLabelIds = [];
+  oldLabels.forEach(label => {
+    OldLabelIds.push(label.id ? label.id : undefined);
+  });
+  newLabels.forEach(label => {
+    if (!OldLabelIds.includes(label.id ? label.id : undefined)) {
+      updatedLabels.push(label);
+    }
+  });
+  return updatedLabels;
+}
+
+function checkForModifications(localConfig, publishedConfig) {
+  const formattedPublishedConfig = labelFormatter(publishedConfig);
+  const labelChanges = [];
+  localConfig.forEach(localLabel => {
+    formattedPublishedConfig.forEach(publicLabel => {
+      if (localLabel.id === publicLabel.id) {
+        if (
+          Object.values(localLabel).toString() !==
+          Object.values(publicLabel).toString()
```

Comment:
This check is a bit fragile, because if the order of the object fields changes (but not the actual values), the condition would be a false positive.

You need to check field by field. In this case, it's also fine to do it "manually", as you only need to check for 3 fields.

Reply:
I understand. This should be a better solution...
d2e48dc